api/client.go

```diff
@@ -60,6 +60,28 @@ type Client interface {
 	Do(context.Context, *http.Request) (*http.Response, []byte, error)
 }
 
+// DoGetFallback will attempt to do the request as-is, and on a 405 it will fallback to a GET request
```

Comment:
missing full stops here and few other places
```suggestion
// DoGetFallback will attempt to do the request as-is, and on a 405 it will fallback to a GET request.
```